pkg/oauthserver/oauthserver/auth.go

```diff
@@ -690,6 +690,10 @@ func (c *OAuthServerConfig) getAuthenticationRequestHandler() (authenticator.Req
 						return nil, fmt.Errorf("Error loading certs from %s: %v", provider.ClientCA, err)
 					}
 
+					// we need to add our CA data to secure serving as well to have the OAuth server
+					// advertise them for client auth during TLS handshake
+					c.GenericConfig.SecureServing.ClientCA = append(c.GenericConfig.SecureServing.ClientCA, provider.ClientCA)
```

Comment:
/hold
This flag's representation changed in later versions, I need to see that this still works

Reply:
/hold cancel
the proper CA names are being announced